src/main/java/step2/domain/lotto/LottoList.java

```diff
@@ -2,12 +2,25 @@
 
 import step2.dto.LottoListDTO;
 
+import java.util.ArrayList;
 import java.util.List;
 import java.util.stream.Collectors;
+import java.util.stream.Stream;
 
 public class LottoList {
     private final List<NormalLotto> lottos;
 
+    public LottoList(ArrayList<String> rawLottoList) {
+        this(rawLottoList.stream()
```

Comment:
```suggestion
    public static LottoList of(List<String> rawLottoList) {
        this(rawLottoList.stream()
```

밑에있는 또다른 `LottoList(List<NormalLotto> lottoList)` 때문에 "same erasure" 에러가 떠서 ArrayList 타입을 활용하시게 된 것 같은데,
차라리 메소드 작명을 통해 의미를 부여할 수 있는 정적 팩터리 메서드를 고려하는 것은 어떨까요?

"엘레강트 오브젝트" 관점에서는 정적 팩터리 메서드라는 존재를 극혐이라며 부정하지만,
조슈아 블로크의 "이펙티브 자바" 에서는 정적 팩터리 메서드의 사용을 나쁘게 생각하지 않습니다.

저는 가능한 기본 생성자를 사용하려고는 하지만,
정적 팩토리 메서드를 나쁘게 생각하는 것보다, 파라미터 타입으로 interface 가 아닌 타입을 넣는 것을 더 싫어하는 편입니다.

실제로 이 때문에, LottoService 에서 `(ArrayList<String>)` 으로 명시적인 타입 캐스팅을 하고 있고요.
이펙티브 자바에서는 불필요한 오토박싱과 타입 캐스팅이 성능상의 오버헤드라는 관점을 취하고 있고,
저는 사실 이펙티브 자바를 "엘레강트 오브젝트" 보다 좋아합니다.

물론 이건 사람에 따라 의견차이가 있을 수 있으니, 그냥 이런 의견도 있구나 하고 넘어가주세요.
꼭 수정해주실 필요는 없습니다.

Reply:
same erasure 에러가 떠서 ArrayList를 사용한 것이 맞습니다ㅠㅠ 두개이니 List와 ArrayList로 처리했지만 만약에 5개라면 Set HashSet이런식으로 꼼수를 썼어야 하는 방법이라 정적 팩토리메소드를 쓰는 것이 확장성의 측면에서 더욱 좋은 것 같습니다. 좋은 피드백 감사합니다!

Reply:
PR을 날리고 나서 든 생각인데 여러 타입의 리스트를 허용해야 하는 경우라면 상위 클래스로 받은 다음에 파싱해서 하나의 주 생성자를 부르는 것이 더 좋은 것 같다는 생각도 드네요.

Reply:
이것도 훌륭한 아이디어네요